freud/plot.py

```diff
@@ -537,7 +546,9 @@ def diffraction_plot(
     )
     ax_divider = make_axes_locatable(ax)
     cax = ax_divider.append_axes("right", size="7%", pad="10%")
-    cb = Colorbar(cax, im)
+    # ticks are placed at top and bottom of color bar
+    cb = plt.colorbar(im, cax=cax, ax=ax, ticks=[vmin, vmax])
```

Comment:
I don't think we want to use the pyplot (`plt`) interface here - its behavior is different than the object-oriented API, I think it adds Axes objects or something weird. You might be able to revert to the previous code but adapt this example to set the ticks? https://matplotlib.org/stable/gallery/ticks_and_spines/colorbar_tick_labelling_demo.html

See #533, #318 for some relevant past work on plotting functionality.